analyzers/tests/SonarAnalyzer.UnitTest/Trackers/ConstantValueFinderTest.cs

```diff
@@ -57,5 +57,33 @@ public int Method()
             var finder = new CSharpConstantValueFinder(compilation.GetSemanticModel(tree));
             finder.FindConstant(returnExpression).Should().Be(42);
         }
+
+        [TestMethod]
+        public void WrongCompilationBeingUsed()
+        {
+            const string firstSnippet = @"
+public class Foo
+{
+    private static int Original = 42;
+    private int Field = Original;
+}";
+            const string secondSnippet = @"
+public class Bar
+{
+    public int Method()
```

Comment:
this code still doesn't compile

So even if you use a secondCompilationFinder, it will still return `null`, because the `Field` symbol does not have any symbol.

If I add these to the test, it fails, because there's no symbol:
```
var secondCompilationFinder = new CSharpConstantValueFinder(secondCompilation.GetSemanticModel(secondCompilation.SyntaxTrees.Single()));
secondCompilationFinder.FindConstant(secondCompilationReturnExpression).Should().NotBeNull();
```

This brings confusion to the test (is it just testing that the `firstCompilationFinder` cannot find symbols because the wrong compilation is used, or because the `secondSnippet` doesn't compile , so it has no symbols even for its own compilation?).

I suggest to make the `Bar` class compilable.

```
public class Bar
{
    private int Field = 42;
    public int Method()
    {
      return Field;
     }
}
```
And then these assertions will both work:
```
firstCompilationFinder.FindConstant(secondCompilationReturnExpression).Should().BeNull();
secondCompilationFinder.FindConstant(secondCompilationReturnExpression).Should().NotBeNull();
```

Reply:
Haven't got this from the beginning. Indeed, those 2 are not compilable!